lib/kafka/client.rb

```diff
@@ -328,12 +328,9 @@ def normalize_seed_brokers(seed_brokers)
       end
       brokers = []
       seed_brokers.each do |connection|
-        if connection =~ /:\/\//
-          u = URI.parse(connection)
-          brokers << "#{u.host}:#{u.port}"
-        else
-          brokers << connection
-        end
+        connection.prepend("kafka://") unless connection =~ /:\/\//
+        u = URI.parse(connection)
+        brokers << { host: u.hostname, port: u.port || 9092 }
```

Comment:
I actually think it would be preferable to just always store the broker addresses as a URI.